bootstrap.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -exu
+
+die () {
+    echo >&2 "$@"
+    exit 1
+}
+
+[ "$#" -eq 1 ] || die "1 argument required, $# provided. Pleas add the project name"
+
+project_name=$1
+
+# install ruby gems
+bundle install
+
+#install node packages
+yarn
+
+# copy secrets from the example adding a new application secret
+application_secret=$(bundle exec rake secret)
```

Comment:
I think this is not necessary, rails uses a tmp file for dev env secrets

`railties/lib/rails/application.rb`
```ruby
    # The secret_key_base is used as the input secret to the application's key generator, which in turn
    # is used to create all MessageVerifiers/MessageEncryptors, including the ones that sign and encrypt cookies.
    #
    # In development and test, this is randomly generated and stored in a
    # temporary file in <tt>tmp/development_secret.txt</tt>.
    #
    # In all other environments, we look for it first in ENV["SECRET_KEY_BASE"],
    # then credentials.secret_key_base, and finally secrets.secret_key_base. For most applications,
    # the correct place to store it is in the encrypted credentials file.
    def secret_key_base
      if Rails.env.development? || Rails.env.test?
        secrets.secret_key_base ||= generate_development_secret
      else
        validate_secret_key_base(
          ENV["SECRET_KEY_BASE"] || credentials.secret_key_base || secrets.secret_key_base
        )
      end
    end
```

Reply:
That makes it much easier, won't there be any problems with deleting the `tmp` folder? I know I do it once in a while so I'm afraid things signed before might be invalid

Reply:
IDK, I guess it's not a problem in dev environment if the session gets expired or something like that

Reply:
I'll leave it, I don't want to find side effects and if anything it serves as a way to see how to generate dynamic configs

Reply:
but isn't the `SECRET_KEY_BASE` env var completely ignored in dev env?

Reply:
I run the script and..

![image](https://user-images.githubusercontent.com/6373536/90519267-30225600-e13e-11ea-87cc-5221acb349ce.png)

Reply:
lol, so let's just remove `SECRET_KEY_BASE:` from `application.yml.example`? I believe it throws an error if I don't set a value

Reply:
Seems like the ENV VAR is directly used here: `config.secret_key_base = ENV['SECRET_KEY_BASE']` in `config/application.rb` .. looks weird to me 😓